include/libopencm3/stm32/g0/adc.h

```diff
@@ -75,37 +97,37 @@
 @{*/
 
 /** CCRDY: Channel Configuration Ready flag */
-#define ADC_ISR_CCRDY				(1 << 13)
+#define ADC_ISR_CCRDY		(1 << 13)
 
 /**@}*/
 
 /** @addtogroup adc_ier
 @{*/
 
 /** CCRDYIE: Channel Configuration Ready Interrupt enable bit */
-#define ADC_IER_CCRDYIE				(1 << 13)
+#define ADC_IER_CCRDYIE		(1 << 13)
 
 /**@}*/
 
 /** @addtogroup adc_ccr
 @{*/
 
-#define ADC_CCR_PRESC_MASK		(0xf)
-#define ADC_CCR_PRESC_SHIFT		(18)
+#define ADC_CCR_PRESC_SHIFT	18
+#define ADC_CCR_PRESC_MASK	(0xf << ADC_CCR_PRESC_SHIFT)
 /** @defgroup adc_ccr_presc ADC clock prescaler
  *@{*/
-#define ADC_CCR_PRESC_NODIV		(0x0)
-#define ADC_CCR_PRESC_DIV1		(0x1)
-#define ADC_CCR_PRESC_DIV2		(0x2)
-#define ADC_CCR_PRESC_DIV6		(0x3)
-#define ADC_CCR_PRESC_DIV8		(0x4)
-#define ADC_CCR_PRESC_DIV10		(0x5)
-#define ADC_CCR_PRESC_DIV12		(0x6)
-#define ADC_CCR_PRESC_DIV16		(0x7)
-#define ADC_CCR_PRESC_DIV32		(0x8)
-#define ADC_CCR_PRESC_DIV64		(0x9)
-#define ADC_CCR_PRESC_DIV128	(0x10)
-#define ADC_CCR_PRESC_DIV256	(0x11)
+#define ADC_CCR_PRESC_NODIV	(0x0 << ADC_CCR_PRESC_SHIFT)
+#define ADC_CCR_PRESC_DIV1	(0x1 << ADC_CCR_PRESC_SHIFT)
+#define ADC_CCR_PRESC_DIV2	(0x2 << ADC_CCR_PRESC_SHIFT)
+#define ADC_CCR_PRESC_DIV6	(0x3 << ADC_CCR_PRESC_SHIFT)
+#define ADC_CCR_PRESC_DIV8	(0x4 << ADC_CCR_PRESC_SHIFT)
+#define ADC_CCR_PRESC_DIV10	(0x5 << ADC_CCR_PRESC_SHIFT)
+#define ADC_CCR_PRESC_DIV12	(0x6 << ADC_CCR_PRESC_SHIFT)
+#define ADC_CCR_PRESC_DIV16	(0x7 << ADC_CCR_PRESC_SHIFT)
+#define ADC_CCR_PRESC_DIV32	(0x8 << ADC_CCR_PRESC_SHIFT)
+#define ADC_CCR_PRESC_DIV64	(0x9 << ADC_CCR_PRESC_SHIFT)
+#define ADC_CCR_PRESC_DIV128	(0x10 << ADC_CCR_PRESC_SHIFT)
+#define ADC_CCR_PRESC_DIV256	(0x11 << ADC_CCR_PRESC_SHIFT)
```

Comment:
nope.  All future work has been using unshifted forms, and has been for some time.  What are you basing this on?  What are you trying to be compatible with?

Reply:
That's a tough one....😕

On one hand, we have shifted forms of `ADC_CFGR2_CKMODE` for F0:
https://github.com/libopencm3/libopencm3/blob/master/include/libopencm3/stm32/f0/adc.h#L86

On the other, we have unshifted forms of `ADC_CFGR2_CKMODE` for G0:
https://github.com/libopencm3/libopencm3/blob/master/include/libopencm3/stm32/g0/adc.h#L140
and for L0:
https://github.com/libopencm3/libopencm3/blob/master/include/libopencm3/stm32/l0/adc.h#L72

So for the same `ADC_CFGR2_CKMODE` basically within the same family (F0/G0/L0) with 99% peripheral compatibility, currently one is forced to write something like this:
```c
#ifdef STM32G0
ADC1_CFGR2 = ADC_CFGR2_CKMODE_PCLK << ADC_CFGR2_CKMODE_SHIFT;
#else
ADC1_CFGR2 = ADC_CFGR2_CKMODE_PCLK;
#endif
```
Isn't it awkward? And that's **only** because of "future work"? Hmm.... weird.😊

Another example - `ADC_CFGR2_OVSR` - shifted in G4:
https://github.com/libopencm3/libopencm3/blob/master/include/libopencm3/stm32/g4/adc.h#L146
unshifted in G0:
https://github.com/libopencm3/libopencm3/blob/master/include/libopencm3/stm32/g0/adc.h#L163

So how does one make the right decision? Pull `g0/adc.h` to the rest of the families (make it shifted) or leave it unshifted because it's "future work"? Since there's no clear documented policy nor is there 100% consistency across the code, I guess one will always do the "wrong" thing unless majority vote (analyze the code and see how other similar things are implemented) and consistency are the main decision making factor. That's exactly what I tried to do in this PR. Otherwise, it stops making any sense.

I feel discouraged to fight "holy wars", so please confirm what shifts are still a problem, and I'll fix them.

Reply:
Btw, the G4 is a total mess.

Shifted mask, shifted values:
https://github.com/libopencm3/libopencm3/blob/master/include/libopencm3/stm32/g4/adc.h#L68

Unshifted mask, shifted values:
https://github.com/libopencm3/libopencm3/blob/master/include/libopencm3/stm32/g4/adc.h#L465

Unshifted values with no mask:
https://github.com/libopencm3/libopencm3/blob/master/include/libopencm3/stm32/g4/adc.h#L188

Go figure...😊